Integration test still wonky: TestMulticlusterServiceDNS fails intermittently

**What happened:** 
go integration test fails intermittently when the pre-commit.sh is run @ build. Test output reports Error updating federated cluster status:
```
   --- FAIL: TestIntegration/Parallel-Integration-Test-Group (0.00s)
        --- FAIL: TestIntegration/Parallel-Integration-Test-Group/TestMultiClusterServiceDNS (0.10s)
        	logger.go:40: Error updating federated cluster status %q: %v test-cluster-pl8dx Operation cannot be fulfilled on federatedclusters.core.federation.k8s.io "test-cluster-pl8dx": the object has been modified; please apply your changes to the latest version and try again
```
<details>
<summary>Parallel-Integration-Test-Group failure message</summary>
 
```
+go test -v ./test/integration ./test/e2e
=== RUN   TestIntegration
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
Flag --insecure-bind-address has been deprecated, This flag will be removed in a future version.
I0710 21:22:57.924040    5193 serving.go:273] Generated self-signed cert (/tmp/k8s_test_framework_174666800/apiserver.crt, /tmp/k8s_test_framework_174666800/apiserver.key)
I0710 21:22:57.924076    5193 server.go:703] external host was not specified, using 10.20.0.59
W0710 21:22:57.924093    5193 authentication.go:378] AnonymousAuth is not allowed with the AlwaysAllow authorizer. Resetting AnonymousAuth to false. You should use a different authorizer
I0710 21:22:57.924546    5193 server.go:145] Version: v1.11.0
I0710 21:22:57.927456    5193 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:22:57.927476    5193 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:22:57.928178    5193 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:22:57.928207    5193 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:22:57.972326    5193 master.go:234] Using reconciler: lease
W0710 21:22:59.806243    5193 genericapiserver.go:319] Skipping API batch/v2alpha1 because it has no resources.
W0710 21:23:00.696809    5193 genericapiserver.go:319] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:00.712215    5193 genericapiserver.go:319] Skipping API scheduling.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:00.762247    5193 genericapiserver.go:319] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:02.495928    5193 genericapiserver.go:319] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/07/10 21:23:02 log.go:33: [restful/swagger] listing is available at http://10.20.0.59:443/swaggerapi
[restful] 2018/07/10 21:23:02 log.go:33: [restful/swagger] http://10.20.0.59:443/swaggerui/ is mapped to folder /swagger-ui/
[restful] 2018/07/10 21:23:05 log.go:33: [restful/swagger] listing is available at http://10.20.0.59:443/swaggerapi
[restful] 2018/07/10 21:23:05 log.go:33: [restful/swagger] http://10.20.0.59:443/swaggerui/ is mapped to folder /swagger-ui/
I0710 21:23:05.602969    5193 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:23:05.603000    5193 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:23:11.902787    5193 insecure_handler.go:119] Serving insecurely on 127.0.0.1:37674
I0710 21:23:11.905624    5193 autoregister_controller.go:136] Starting autoregister controller
I0710 21:23:11.905640    5193 cache.go:32] Waiting for caches to sync for autoregister controller
I0710 21:23:11.905671    5193 apiservice_controller.go:90] Starting APIServiceRegistrationController
I0710 21:23:11.905678    5193 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I0710 21:23:11.905696    5193 available_controller.go:278] Starting AvailableConditionController
I0710 21:23:11.905702    5193 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I0710 21:23:11.905720    5193 controller.go:84] Starting OpenAPI AggregationController
I0710 21:23:11.907204    5193 crd_finalizer.go:242] Starting CRDFinalizer
I0710 21:23:11.907231    5193 crdregistration_controller.go:112] Starting crd-autoregister controller
I0710 21:23:11.907250    5193 controller_utils.go:1025] Waiting for caches to sync for crd-autoregister controller
I0710 21:23:11.909704    5193 customresource_discovery_controller.go:199] Starting DiscoveryController
I0710 21:23:11.909729    5193 naming_controller.go:284] Starting NamingConditionController
I0710 21:23:11.909747    5193 establishing_controller.go:73] Starting EstablishingController
W0710 21:23:11.958890    5193 lease.go:223] Resetting endpoints for master service "kubernetes" to [10.20.0.59]
I0710 21:23:12.005792    5193 cache.go:39] Caches are synced for AvailableConditionController controller
I0710 21:23:12.006084    5193 cache.go:39] Caches are synced for autoregister controller
I0710 21:23:12.006683    5193 cache.go:39] Caches are synced for APIServiceRegistrationController controller
I0710 21:23:12.009546    5193 controller_utils.go:1032] Caches are synced for crd-autoregister controller
I0710 21:23:12.914674    5193 storage_scheduling.go:91] created PriorityClass system-node-critical with value 2000001000
I0710 21:23:12.917027    5193 storage_scheduling.go:91] created PriorityClass system-cluster-critical with value 2000000000
I0710 21:23:12.917043    5193 storage_scheduling.go:100] all system priority classes are created successfully or already exist.
2018/07/10 21:23:12 Creating CRD clusters.clusterregistry.k8s.io
2018/07/10 21:23:12 Finished installing.
2018/07/10 21:23:13 Creating CRD federatedclusters.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federatedconfigmaps.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federatedconfigmapoverrides.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federatedconfigmapplacements.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federateddeployments.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federateddeploymentoverrides.core.federation.k8s.io
2018/07/10 21:23:13 Creating CRD federateddeploymentplacements.core.federation.k8s.io
2018/07/10 21:23:14 Creating CRD federatedjobs.core.federation.k8s.io
2018/07/10 21:23:14 Creating CRD federatedjoboverrides.core.federation.k8s.io
2018/07/10 21:23:15 Creating CRD federatedjobplacements.core.federation.k8s.io
2018/07/10 21:23:15 Creating CRD federatednamespaceplacements.core.federation.k8s.io
2018/07/10 21:23:15 Creating CRD federatedreplicasets.core.federation.k8s.io
2018/07/10 21:23:16 Creating CRD federatedreplicasetoverrides.core.federation.k8s.io
2018/07/10 21:23:16 Creating CRD federatedreplicasetplacements.core.federation.k8s.io
2018/07/10 21:23:17 Creating CRD federatedsecrets.core.federation.k8s.io
2018/07/10 21:23:17 Creating CRD federatedsecretoverrides.core.federation.k8s.io
2018/07/10 21:23:17 Creating CRD federatedsecretplacements.core.federation.k8s.io
2018/07/10 21:23:18 Creating CRD federatedservices.core.federation.k8s.io
2018/07/10 21:23:18 Creating CRD federatedserviceplacements.core.federation.k8s.io
2018/07/10 21:23:19 Creating CRD federatedtypeconfigs.core.federation.k8s.io
2018/07/10 21:23:19 Creating CRD propagatedversions.core.federation.k8s.io
2018/07/10 21:23:19 Creating CRD multiclusterservicednsrecords.multiclusterdns.federation.k8s.io
2018/07/10 21:23:20 Creating CRD replicaschedulingpreferences.scheduling.federation.k8s.io
2018/07/10 21:23:20 Finished installing.
E0710 21:23:20.597757    5193 autoregister_controller.go:190] v1alpha1.scheduling.federation.k8s.io failed with : apiservices.apiregistration.k8s.io "v1alpha1.scheduling.federation.k8s.io" already exists
I0710 21:23:20.667712    5193 controller.go:597] quota admission added evaluator for: {clusterregistry.k8s.io clusters}
I0710 21:23:20.681110    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedclusters}
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
Flag --insecure-bind-address has been deprecated, This flag will be removed in a future version.
I0710 21:23:21.107367    5246 serving.go:273] Generated self-signed cert (/tmp/k8s_test_framework_717007311/apiserver.crt, /tmp/k8s_test_framework_717007311/apiserver.key)
I0710 21:23:21.107396    5246 server.go:703] external host was not specified, using 10.20.0.59
W0710 21:23:21.107412    5246 authentication.go:378] AnonymousAuth is not allowed with the AlwaysAllow authorizer. Resetting AnonymousAuth to false. You should use a different authorizer
I0710 21:23:21.107810    5246 server.go:145] Version: v1.11.0
I0710 21:23:21.110595    5246 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:23:21.110613    5246 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:23:21.111283    5246 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:23:21.111311    5246 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:23:21.152889    5246 master.go:234] Using reconciler: lease
W0710 21:23:22.668721    5246 genericapiserver.go:319] Skipping API batch/v2alpha1 because it has no resources.
W0710 21:23:23.626387    5246 genericapiserver.go:319] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:23.641863    5246 genericapiserver.go:319] Skipping API scheduling.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:23.764262    5246 genericapiserver.go:319] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0710 21:23:25.474639    5246 genericapiserver.go:319] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/07/10 21:23:25 log.go:33: [restful/swagger] listing is available at http://10.20.0.59:443/swaggerapi
[restful] 2018/07/10 21:23:25 log.go:33: [restful/swagger] http://10.20.0.59:443/swaggerui/ is mapped to folder /swagger-ui/
[restful] 2018/07/10 21:23:28 log.go:33: [restful/swagger] listing is available at http://10.20.0.59:443/swaggerapi
[restful] 2018/07/10 21:23:28 log.go:33: [restful/swagger] http://10.20.0.59:443/swaggerui/ is mapped to folder /swagger-ui/
I0710 21:23:28.607584    5246 plugins.go:158] Loaded 6 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0710 21:23:28.607617    5246 plugins.go:161] Loaded 5 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0710 21:23:34.922421    5246 insecure_handler.go:119] Serving insecurely on 127.0.0.1:41816
I0710 21:23:34.925136    5246 apiservice_controller.go:90] Starting APIServiceRegistrationController
I0710 21:23:34.925164    5246 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I0710 21:23:34.925197    5246 autoregister_controller.go:136] Starting autoregister controller
I0710 21:23:34.925204    5246 cache.go:32] Waiting for caches to sync for autoregister controller
I0710 21:23:34.925226    5246 available_controller.go:278] Starting AvailableConditionController
I0710 21:23:34.925232    5246 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I0710 21:23:34.926170    5246 crd_finalizer.go:242] Starting CRDFinalizer
I0710 21:23:34.927231    5246 controller.go:84] Starting OpenAPI AggregationController
I0710 21:23:34.927279    5246 crdregistration_controller.go:112] Starting crd-autoregister controller
I0710 21:23:34.927298    5246 controller_utils.go:1025] Waiting for caches to sync for crd-autoregister controller
I0710 21:23:34.927352    5246 customresource_discovery_controller.go:199] Starting DiscoveryController
I0710 21:23:34.927373    5246 naming_controller.go:284] Starting NamingConditionController
I0710 21:23:34.927390    5246 establishing_controller.go:73] Starting EstablishingController
W0710 21:23:34.960749    5246 lease.go:223] Resetting endpoints for master service "kubernetes" to [10.20.0.59]
I0710 21:23:35.025341    5246 cache.go:39] Caches are synced for AvailableConditionController controller
I0710 21:23:35.025380    5246 cache.go:39] Caches are synced for autoregister controller
I0710 21:23:35.025341    5246 cache.go:39] Caches are synced for APIServiceRegistrationController controller
I0710 21:23:35.027757    5246 controller_utils.go:1032] Caches are synced for crd-autoregister controller
I0710 21:23:35.929305    5246 storage_scheduling.go:91] created PriorityClass system-node-critical with value 2000001000
I0710 21:23:35.931787    5246 storage_scheduling.go:91] created PriorityClass system-cluster-critical with value 2000000000
I0710 21:23:35.931803    5246 storage_scheduling.go:100] all system priority classes are created successfully or already exist.
=== RUN   TestIntegration/Parallel-Integration-Test-Group
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud
=== PAUSE TestIntegration/Parallel-Integration-Test-Group/TestCrud
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference
=== PAUSE TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestMultiClusterServiceDNS
=== PAUSE TestIntegration/Parallel-Integration-Test-Group/TestMultiClusterServiceDNS
=== CONT  TestIntegration/Parallel-Integration-Test-Group/TestCrud
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedConfigMap
=== CONT  TestIntegration/Parallel-Integration-Test-Group/TestMultiClusterServiceDNS
I0710 21:23:36.021530    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedconfigmaps}
E0710 21:23:36.037120    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 118 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[]  }} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.043373    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-pl8dx  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-pl8dx 8150688c-8487-11e8-acb8-42010a14003b 159 1 2018-07-10 21:23:35 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-pl8dx} &LocalObjectReference{Name:test-cluster-pl8dx-credentialsrnbdl,}} {[]  }} not added.  Not of correct type, or cluster not ready.
I0710 21:23:36.057669    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedconfigmapplacements}
E0710 21:23:36.071488    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-pl8dx  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-pl8dx 8150688c-8487-11e8-acb8-42010a14003b 159 1 2018-07-10 21:23:35 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-pl8dx} &LocalObjectReference{Name:test-cluster-pl8dx-credentialsrnbdl,}} {[]  }} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.071651    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 118 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[]  }} not added.  Not of correct type, or cluster not ready.
I0710 21:23:36.066522    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedconfigmapoverrides}
E0710 21:23:36.090515    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 163 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[] zoneht65g regionht65g}} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.090615    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-pl8dx  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-pl8dx 8150688c-8487-11e8-acb8-42010a14003b 159 1 2018-07-10 21:23:35 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-pl8dx} &LocalObjectReference{Name:test-cluster-pl8dx-credentialsrnbdl,}} {[]  }} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.091655    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 118 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[]  }} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.091726    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-pl8dx  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-pl8dx 8150688c-8487-11e8-acb8-42010a14003b 159 1 2018-07-10 21:23:35 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-pl8dx} &LocalObjectReference{Name:test-cluster-pl8dx-credentialsrnbdl,}} {[]  }} not added.  Not of correct type, or cluster not ready.
I0710 21:23:36.092470    5193 controller.go:597] quota admission added evaluator for: { namespaces}
=== CONT  TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference
E0710 21:23:36.133407    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 163 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[] zoneht65g regionht65g}} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.146808    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 163 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[] zoneht65g regionht65g}} not added.  Not of correct type, or cluster not ready.
E0710 21:23:36.173664    5174 federated_informer.go:209] Cluster &{{FederatedCluster core.federation.k8s.io/v1alpha1} {test-cluster-ht65g  federation-system /apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters/test-cluster-ht65g 78323940-8487-11e8-acb8-42010a14003b 163 1 2018-07-10 21:23:20 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{test-cluster-ht65g} &LocalObjectReference{Name:test-cluster-ht65g-credentialscf45s,}} {[] zoneht65g regionht65g}} not added.  Not of correct type, or cluster not ready.
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_equally_in_clusters,_with_no_explicit_per_cluster_preferences
I0710 21:23:36.199625    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federateddeployments}
I0710 21:23:36.215334    5193 controller.go:597] quota admission added evaluator for: {scheduling.federation.k8s.io replicaschedulingpreferences}
I0710 21:23:36.220535    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io propagatedversions}
I0710 21:23:36.248269    5246 controller.go:597] quota admission added evaluator for: { namespaces}
I0710 21:23:37.226961    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federateddeploymentplacements}
I0710 21:23:37.232298    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federateddeploymentoverrides}
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_in_proportion_of_weights_when_explicit_preferences_with_weights_specified
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_considering_min_replicas_when_both_minreplica_and_weights_specified
E0710 21:23:38.691675    5174 controller.go:447] Failed to delete FederatedConfigMap "crud-federatedconfigmap-l6szd/test-crud-fxmqg": waiting for object crud-federatedconfigmap-l6szd/test-crud-fxmqg to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedDeployment
I0710 21:23:38.963196    5246 controller.go:597] quota admission added evaluator for: {apps deployments}
I0710 21:23:38.971263    5193 controller.go:597] quota admission added evaluator for: {apps deployments}
E0710 21:23:39.696064    5174 controller.go:447] Failed to delete FederatedDeployment "crud-federateddeployment-qq67s/test-crud-sg8gn": waiting for object crud-federateddeployment-qq67s/test-crud-sg8gn to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedJob
I0710 21:23:39.903757    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedjobs}
I0710 21:23:39.920462    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedjobplacements}
I0710 21:23:39.926154    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedjoboverrides}
I0710 21:23:39.945702    5246 controller.go:597] quota admission added evaluator for: {batch jobs}
E0710 21:23:39.967910    5174 controller.go:616] Failed to record propagated version for Namespace "crud-federatedjob-j2rqq": Operation cannot be fulfilled on propagatedversions.core.federation.k8s.io "namespace-crud-federatedjob-j2rqq": the object has been modified; please apply your changes to the latest version and try again
I0710 21:23:39.970064    5193 controller.go:597] quota admission added evaluator for: {batch jobs}
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_in_proportion_of_weights_when_explicit_preferences_with_weights_specified#01
I0710 21:23:40.313639    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedreplicasets}
E0710 21:23:41.314703    5174 controller.go:447] Failed to delete FederatedJob "crud-federatedjob-j2rqq/test-crud-gg4x5": waiting for object crud-federatedjob-j2rqq/test-crud-gg4x5 to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedReplicaSet
I0710 21:23:41.538630    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedreplicasetplacements}
I0710 21:23:41.548762    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedreplicasetoverrides}
I0710 21:23:41.563266    5246 controller.go:597] quota admission added evaluator for: {apps replicasets}
I0710 21:23:41.582539    5193 controller.go:597] quota admission added evaluator for: {apps replicasets}
E0710 21:23:41.594881    5174 controller.go:616] Failed to record propagated version for Namespace "crud-federatedreplicaset-pdskx": Operation cannot be fulfilled on propagatedversions.core.federation.k8s.io "namespace-crud-federatedreplicaset-pdskx": the object has been modified; please apply your changes to the latest version and try again
E0710 21:23:42.139260    5174 controller.go:447] Failed to delete FederatedReplicaSet "crud-federatedreplicaset-pdskx/test-crud-vvmv2": waiting for object crud-federatedreplicaset-pdskx/test-crud-vvmv2 to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedSecret
I0710 21:23:42.355472    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedsecrets}
I0710 21:23:42.369887    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedsecretplacements}
I0710 21:23:42.374653    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedsecretoverrides}
E0710 21:23:42.394118    5174 controller.go:616] Failed to record propagated version for Namespace "crud-federatedsecret-vhpzb": Operation cannot be fulfilled on propagatedversions.core.federation.k8s.io "namespace-crud-federatedsecret-vhpzb": the object has been modified; please apply your changes to the latest version and try again
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_considering_min_replicas_when_both_minreplica_and_weights_specified#01
E0710 21:23:43.146771    5174 controller.go:447] Failed to delete FederatedSecret "crud-federatedsecret-vhpzb/test-crud-drp5j": waiting for object crud-federatedsecret-vhpzb/test-crud-drp5j to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestCrud/FederatedService
I0710 21:23:43.350625    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedservices}
I0710 21:23:43.355992    5193 controller.go:597] quota admission added evaluator for: {core.federation.k8s.io federatedserviceplacements}
E0710 21:23:44.979761    5174 controller.go:447] Failed to delete FederatedService "crud-federatedservice-qssdd/test-crud-dc6j5": waiting for object crud-federatedservice-qssdd/test-crud-dc6j5 to be deleted from clusters: test-cluster-ht65g
=== RUN   TestIntegration/Parallel-Integration-Test-Group/TestReplicaSchedulingPreference/Replicas_spread_equally_in_clusters,_with_no_explicit_per_cluster_preferences#01
I0710 21:23:47.414872    5193 naming_controller.go:295] Shutting down NamingConditionController
I0710 21:23:47.414906    5193 customresource_discovery_controller.go:210] Shutting down DiscoveryController
I0710 21:23:47.414923    5193 establishing_controller.go:84] Shutting down EstablishingController
I0710 21:23:47.414937    5193 crd_finalizer.go:254] Shutting down CRDFinalizer
I0710 21:23:47.414953    5193 crdregistration_controller.go:143] Shutting down crd-autoregister controller
I0710 21:23:47.414968    5193 apiservice_controller.go:102] Shutting down APIServiceRegistrationController
I0710 21:23:47.414986    5193 autoregister_controller.go:160] Shutting down autoregister controller
I0710 21:23:47.415002    5193 available_controller.go:290] Shutting down AvailableConditionController
I0710 21:23:47.415181    5193 controller.go:90] Shutting down OpenAPI AggregationController
I0710 21:23:47.415246    5193 controller.go:158] Shutting down kubernetes service endpoint reconciler
I0710 21:23:47.416208    5193 serve.go:136] Stopped listening on 127.0.0.1:37674
E0710 21:23:47.418708    5193 controller.go:160] no master IPs were listed in storage, refusing to erase all endpoints for the kubernetes service
E0710 21:23:47.431452    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:420: Failed to watch <nil>: Get http://127.0.0.1:37674/api/v1/services?resourceVersion=370&timeoutSeconds=586&watch=true: dial tcp 127.0.0.1:37674: connect: connection refused
E0710 21:23:47.431549    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:420: Failed to watch <nil>: Get http://127.0.0.1:37674/api/v1/endpoints?resourceVersion=8&timeoutSeconds=578&watch=true: dial tcp 127.0.0.1:37674: connect: connection refused
E0710 21:23:47.432164    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:310: Failed to watch *v1alpha1.FederatedCluster: Get http://127.0.0.1:37674/apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters?resourceVersion=385&timeoutSeconds=513&watch=true: dial tcp 127.0.0.1:37674: connect: connection refused
E0710 21:23:47.432222    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:310: Failed to watch *v1alpha1.FederatedCluster: Get http://127.0.0.1:37674/apis/core.federation.k8s.io/v1alpha1/namespaces/federation-system/federatedclusters?resourceVersion=385&timeoutSeconds=404&watch=true: dial tcp 127.0.0.1:37674: connect: connection refused
E0710 21:23:47.432371    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/servicedns/controller.go:201: Failed to watch *v1alpha1.MultiClusterServiceDNSRecord: Get http://127.0.0.1:37674/apis/multiclusterdns.federation.k8s.io/v1alpha1/multiclusterservicednsrecords?resourceVersion=115&timeoutSeconds=507&watch=true: dial tcp 127.0.0.1:37674: connect: connection refused
E0710 21:23:47.512710    5174 scheduler.go:206] Failed to reconcile Federation Targets for RSP named "rsp-57gvq/test-replicaset-6l22s": Get http://127.0.0.1:37674/apis/core.federation.k8s.io/v1alpha1/namespaces/rsp-57gvq/federatedreplicasetoverrides/test-replicaset-6l22s: dial tcp 127.0.0.1:37674: connect: connection refused
I0710 21:23:47.536906    5246 naming_controller.go:295] Shutting down NamingConditionController
I0710 21:23:47.536938    5246 customresource_discovery_controller.go:210] Shutting down DiscoveryController
I0710 21:23:47.536953    5246 establishing_controller.go:84] Shutting down EstablishingController
I0710 21:23:47.536967    5246 crd_finalizer.go:254] Shutting down CRDFinalizer
I0710 21:23:47.536982    5246 crdregistration_controller.go:143] Shutting down crd-autoregister controller
I0710 21:23:47.536997    5246 apiservice_controller.go:102] Shutting down APIServiceRegistrationController
I0710 21:23:47.537014    5246 autoregister_controller.go:160] Shutting down autoregister controller
I0710 21:23:47.537029    5246 available_controller.go:290] Shutting down AvailableConditionController
I0710 21:23:47.537499    5246 controller.go:90] Shutting down OpenAPI AggregationController
I0710 21:23:47.537543    5246 controller.go:158] Shutting down kubernetes service endpoint reconciler
I0710 21:23:47.537700    5246 serve.go:136] Stopped listening on 127.0.0.1:41816
E0710 21:23:47.540354    5246 controller.go:160] no master IPs were listed in storage, refusing to erase all endpoints for the kubernetes service
E0710 21:23:47.551675    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:420: Failed to watch <nil>: Get http://127.0.0.1:41816/api/v1/endpoints?resourceVersion=127&timeoutSeconds=469&watch=true: dial tcp 127.0.0.1:41816: connect: connection refused
E0710 21:23:47.551914    5174 reflector.go:322] github.com/kubernetes-sigs/federation-v2/pkg/controller/util/federated_informer.go:420: Failed to watch <nil>: Get http://127.0.0.1:41816/api/v1/services?resourceVersion=365&timeoutSeconds=494&watch=true: dial tcp 127.0.0.1:41816: connect: connection refused
--- FAIL: TestIntegration (51.22s)
	logger.go:52: Starting a federation of 2 clusters...
	logger.go:52: Creating Cluster Registry CRD
	logger.go:52: Creating Federation CRDs
	logger.go:52: Added cluster test-cluster-ht65g to the federation
	logger.go:52: Added cluster test-cluster-pl8dx to the federation
	logger.go:52: Starting cluster controller
	logger.go:48: Federation started.
    --- FAIL: TestIntegration/Parallel-Integration-Test-Group (0.00s)
        --- FAIL: TestIntegration/Parallel-Integration-Test-Group/TestMultiClusterServiceDNS (0.10s)
        	logger.go:40: Error updating federated cluster status %q: %v test-cluster-pl8dx Operation cannot be fulfilled on federatedclusters.core.federation.k8s.io "test-cluster-pl8dx": the object has been modified; please apply your changes to the latest version and try again
```
<\details>

--- a/test/integration/servicedns_test.go
+++ b/test/integration/servicedns_test.go
@@ -173,15 +173,7 @@ func newServiceDNSTestFixture(tl common.TestLogger, fedFixture *framework.Federa
 		}
 		f.clusterRegionZones[clusterName] = regionZones
 
-		federatedCluster, err := f.client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).Get(clusterName, metav1.GetOptions{})
-		if err != nil {
-			tl.Fatal("Error retrieving federated cluster %q: %v", clusterName, err)
-		}
-		federatedCluster.Status = f.clusterRegionZones[clusterName]
-		_, err = f.client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).UpdateStatus(federatedCluster)
-		if err != nil {
-			tl.Fatal("Error updating federated cluster status %q: %v", clusterName, err)
-		}
+		updateFederatedClusterStatus(tl, f.client, clusterName, f.clusterRegionZones[clusterName])
 
 		clusterDNS := dnsv1a1.ClusterDNS{
 			Cluster: clusterName,
@@ -203,4 +195,28 @@ func newServiceDNSTestFixture(tl common.TestLogger, fedFixture *framework.Federa
 	})
 
 	return f
+}
+
+func updateFederatedClusterStatus(tl common.TestLogger, client fedclientset.Interface, clusterName string, clusterStatus fedv1a1.FederatedClusterStatus) {
+	err := wait.PollImmediate(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (exist bool, err error) {
+		federatedCluster, err := client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).Get(clusterName, metav1.GetOptions{})
+		if err != nil {
+			tl.Logf("Error retrieving federated cluster %q: %v", clusterName, err)
+			return false, err
+		}
+
+		// Update just the Region & Zone fields in Status instead of writing complete structure
+		// so as not to overwrite other fields in Status of existing object.
+		federatedCluster.Status.Region = clusterStatus.Region
+		federatedCluster.Status.Zone = clusterStatus.Zone
+		_, err = client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).UpdateStatus(federatedCluster)
+		if err != nil {
+			tl.Logf("Retry updating federated cluster status %q: %v", clusterName, err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		tl.Fatalf("Error updating status for cluster %q:%v", clusterName, err)
+	}
 }

